### PR TITLE
CORTX-30330: Resource Quota Update

### PR DESCRIPTION
**Pull request by abhijit1patil**
# Pull Request
## Problem Statement
-   Need to have only 1 session per user
-  Update formulae to update calculation of request quota

## Design
-   At the time of successful login, check if valid session exists, if yes, return that session or create a new session and return
- Formulae :- (quota =  MaxAvailableMemForRequest/ RequestSize)

## Custom build and deployment
- [Custom Build job](https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6689/)
- Custom Build
cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-6689-custom-ci
cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6689-custom-ci
cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6689-custom-ci

-  [ Deployment Job](https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/)

## Dev Testing

Admin Login from Client Machine
image

Admin Login from Postman
image

-   \[ \] Coding conventions are followed and code is consistent

-   \[ \] Unit and System Tests are added
-   \[ \] Test Cases cover Happy Path, Non-Happy Path and Scalability
-   \[ \] Testing was performed with RPM

-   \[ \] Interface change (if any) are documented
-   \[ \] Side effects on other features (deployment/upgrade)
-   \[ \] Dependencies on other component(s)

-   \[ \] JIRA number/GitHub Issue added to PR
-   \[ \] PR is self reviewed
-   \[ \] Jira and state/status is updated and JIRA is updated with PR link
-   \[ \] Check if the description is clear and explained

-   \[ \] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: csm/core/services/sessions.py [hunk 1]
```diff
@@ -410,3 +414,30 @@ async def delete_all_sessions_for_user(self, user_id: str) -> None:
...
+        session.expiry_time = self._session_manager.calc_expiry_time()
+        await self._session_manager.update(session)
+
+    async def _is_valid_session_exists(self, user_id: str) -> None:
+        """
+        This Function will get the current user's active session.
+        :param user_id: user_id for user. :type:Str
+        :return: None
+        """
+        Log.debug(f"get active sessions for Userid: {user_id}")
```
Comment (pawankumarsrivastava) [inline csm/core/services/sessions.py:428]: get active sessions for Userid -> Getting active session for user : {user_id}

File: csm/core/services/sessions.py [hunk 2]
```diff
@@ -410,3 +414,30 @@ async def delete_all_sessions_for_user(self, user_id: str) -> None:
...
+        session_data = await self._session_manager.get_all()
```
Comment (pawankumarsrivastava) [inline csm/core/services/sessions.py:429]: One user will have only one session. Why are you getting all?

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/core/services/sessions.py (hunk 2 above)
Comment (abhijit1patil) [reply csm/core/services/sessions.py:429]: In consul, sessions are stored in mapping of session_id to session model and we want to get sessions with user id
Algo: 
1. get all sessions() -> session list
2. traverse over all session list and Check user id in that each session, if user id matches then check if it is expired, if yes delete it else return the same session

File: csm/core/services/sessions.py (hunk 1 above)
Comment (abhijit1patil) [reply csm/core/services/sessions.py]: updated

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**